Moblin/Various/Model.swift

```diff
@@ -4966,13 +4980,50 @@ extension Model {
         remoteControlAssistant?.getSettings { settings in
             self.remoteControlSettings = settings
         }
+        if database.watch!.remoteControl! {
+            var thermalState = ProcessInfo.processInfo.thermalState
+            switch remoteControlGeneral?.flame {
+            case .white:
+                thermalState = .fair
+            case .yellow:
+                thermalState = .serious
+            case .red:
+                thermalState = .critical
+            case .none:
+                thermalState = .critical
+            }
+            let audioLevel = parseAudioLevel(remoteControlTopRight?.audioLevel?.message)
+            if let recordingMessage = remoteControlTopRight?.recording?.message {
+                sendRecordingLengthToWatch(recordingLength: recordingMessage)
+            }
+            if let bitrateMessage = remoteControlTopRight?.bitrate?.message {
+                sendSpeedAndTotalToWatch(speedAndTotal: bitrateMessage)
+            }
+            sendThermalStateToWatch(thermalState: thermalState)
+            sendAudioLevelToWatch(audioLevel: audioLevel)
+            sendIsLiveToWatch(isLive: remoteControlTopRight?.bitrate?.message != nil)
+            sendIsRecordingToWatch(isRecording: remoteControlTopRight?.recording?.message != nil)
+            sendIsMutedToWatch(isMuteOn: audioLevel.isNaN)
+        }
```

Comment:
Should this be sent when the remote control assistant receives updates, not send the previous update?

Reply:
I don't quite understand. The listed conditions are necessary to set the toggles.

Reply:
`updateRemoteControlAssistantStatus()` requests an update from the streamer. It's the completion blocks for `getStatus()` and `getSettings()` that receives the responses, and it is in those completion blocks we should update the watch.

Reply:
hope it's okay now :)